### PR TITLE
The max file preview size is loaded from the cfg

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java [hunk 1]
```diff
@@ -347,28 +352,22 @@ public String extractFile(InputStream inputStream, String fileType) {
...
-                    sb.append("...too many files...|0");
-                    sb.append("</element>");
-                    break;
-                }
+                sb.append("<element>").append(filePath).append("</element>");
             }
         }
+
+        if (fileCounter > maxPreviewCount) {
+            sb.append("<element>...too many files...|0</element>");
```
Comment (MajoBerger) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java:369]: what is this `|0` at the end? seems out of place

**Review by MajoBerger — CHANGES_REQUESTED** [review]
seems like a typo

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java (hunk 1 above)
Comment (milanmajchrak) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java:369]: Yes, but that `|0` has a higher purpose, that value is converted to Bytes info in the FE.
The result looks like this: 
image

The image is created in the v5

Comment (MajoBerger) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java:369]: that is absolutely awful.. why is it in its own folder?
and it is not really true it has 0 bytes, is it :) 

But if you are ok with it, I don't want to postpone this even more. 
But I DO NOT like it.

**Review by MajoBerger — DISMISSED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java [hunk 2]
```diff
@@ -347,28 +352,22 @@ public String extractFile(InputStream inputStream, String fileType) {
...
         sb.append(("<root>"));
-        List<String> allFiles = filePaths;
+        Iterator<String> iterator = filePaths.iterator();
         int fileCounter = 0;
-        for (String filePath : allFiles) {
-            if (!filePath.isEmpty() && filePath.length() > 3) {
+        while ((iterator.hasNext() && fileCounter < maxPreviewCount)) {
+            String filePath = iterator.next();
+
+            if (filePath.length() > 3) {
```
Comment (vidiecan) [inline dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java:360]: explain this plz

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java (hunk 2 above)
Comment (milanmajchrak) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java:360]: I do not know exactly, this is part of Lock's logic. Should I dig into it? This issue wasn't about this lines of code.

Comment (milanmajchrak) [reply dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataBitstreamRestRepository.java]: Added comments and refactored

**Pull request merged**